code/modules/reagents/chemistry/reagents/blob_reagents.dm

```diff
@@ -10,6 +10,8 @@
 	var/message_living = null //extension to first mob sent to only living mobs i.e. silicons have no skin to be burnt
 
 /datum/reagent/blob/reaction_mob(mob/living/M, method=TOUCH, reac_volume, show_message, touch_protection, mob/camera/blob/O)
+	if(M.stat == DEAD)
+		return 0 //the dead don't cause reactions
```

Comment:
reaction_mob() still occurs, as that's the "reagents touching the skin"

on_mob_life() is what doesn't happen while dead